src/esp/assets/managers/SceneAttributesManager.cpp

```diff
@@ -20,21 +16,172 @@ namespace managers {
 PhysicsSceneAttributes::ptr SceneAttributesManager::createAttributesTemplate(
     const std::string& sceneAttributesHandle,
     bool registerTemplate) {
+  PhysicsSceneAttributes::ptr attrs;
+  std::string msg;
+  std::string strHandle = Cr::Utility::String::lowercase(sceneAttributesHandle);
+  bool fileExists = (this->isValidFileName(sceneAttributesHandle));
+
+  if (fileExists) {
+    if ((strHandle.find(".json") != std::string::npos) && fileExists) {
+      // check if sceneAttributesHandle corresponds to an actual, existing json
+      // scene file descriptor.
+      attrs = createFileBasedAttributesTemplate(sceneAttributesHandle,
+                                                registerTemplate);
+      msg = "JSON File (" + sceneAttributesHandle + ") Based";
+    } else {
+      // if name is not json file descriptor but still appropriate file
+      attrs = createFileBasedAttributesTemplate(sceneAttributesHandle,
+                                                registerTemplate);
+      msg = "File (" + sceneAttributesHandle + ") Based";
+    }
+
+  } else {
+    // if name is not file descriptor, return default attributes.
+    attrs = createDefaultAttributesTemplate(sceneAttributesHandle,
+                                            registerTemplate);
+    msg = "New default";
+  }
+
+  if (nullptr != attrs) {
+    LOG(INFO) << msg << " scene attributes created"
+              << (registerTemplate ? " and registered." : ".");
+  }
+  return attrs;
+
+}  // SceneAttributesManager::createAttributesTemplate
+
+int SceneAttributesManager::registerAttributesTemplateFinalize(
+    PhysicsSceneAttributes::ptr sceneAttributesTemplate,
+    const std::string& sceneAttributesHandle) {
+  if (sceneAttributesTemplate->getRenderAssetHandle() == "") {
+    LOG(ERROR)
+        << "SceneAttributesManager::registerAttributesTemplateFinalize : "
+           "Attributes template named"
+        << sceneAttributesHandle
+        << "does not have a valid render asset handle specified. Aborting.";
+    return ID_UNDEFINED;
+  }
+
+  // Handles for rendering and collision assets
+  std::string renderAssetHandle =
+      sceneAttributesTemplate->getRenderAssetHandle();
+  std::string collisionAssetHandle =
+      sceneAttributesTemplate->getCollisionAssetHandle();
+
+  // verify these represent legitimate assets
```

Comment:
Lets add primitives as valid render/collision assets for a scene. This will support physics simulation without a scene mesh (e.g. primitive cube plane for locomotion experiments).

Reply:
This necessitated some changes to createAttributesTemplate for scene attributes as well.